web/packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -858,8 +863,12 @@ export default class Codec {
   // decodeStringMessage decodes a tdp message of the form
   // | message type (N) | message_length uint32 | message []byte
   private decodeStringMessage(buffer: ArrayBuffer): string {
-    const offset = byteLength + uint32Length; // eat message type and message_length
-    return this.decoder.decode(new Uint8Array(buffer.slice(offset)));
+    const dv = new DataView(buffer);
+    let offset = byteLength; // eat message type
+    const msgLength = dv.getUint32(offset);
```

Comment:
This has been a bug for a loooong time. We ignore the length that the server sends and just assume the rest of the message is one giant string. (Which is why we often see unprintable characters at the end of the error message).